### PR TITLE
feat: parsed smart contract metadata UI

### LINKED ISSUE #5242: [Task]: Show parsed metadata in the UI
### Task description

We should correctly show the parsed metadata for smart contract function calls. The following fields have to be added in a separate smart contract tab:

- Target Contract
- Contract Function
- Gas Budget

### Requirements

- [x] #5216 
- [x] Tests

### Creation checklist

- [x] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by jeeanribeiro**
## Summary
Adds "Smart Contract" tab in ActivityInformation with parsed smart contract metadata

## Changelog

```
- Adds getDestinationNetworkFromAddress util
- Adds parseLayer2Metadata util
- Adds smart contract related strings
- Adds destinationNetwork in BaseActivityType and generateActivity function
- Adds SmartContractActivityInformation component
- Add Smart Contract tab in ActivityInformation
```

## Relevant Issues
closes #5242

### Platforms
- __Desktop__
  - [ ] MacOS
  - [x] Linux
  - [ ] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

## Checklist
- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/components/molecules/activity-info/SmartContractActivityInformation.svelte [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        return {
+            ...(metadata?.targetContract && {
+                targetContract: { data: metadata.targetContract },
+            }),
+            ...(metadata?.contractFunction && {
+                contractFunction: { data: metadata.contractFunction },
+            }),
+            ...(metadata?.gasBudget && {
+                gasBudget: { data: metadata.gasBudget },
+            }),
```
Comment (MarkNerdi) [inline packages/shared/components/molecules/activity-info/SmartContractActivityInformation.svelte:22]: i think the ethereum address would also be relevant here

File: packages/shared/components/molecules/activity-info/SmartContractActivityInformation.svelte [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<script lang="typescript">
+    import { KeyValueBox } from 'shared/components'
+    import { localize } from '@core/i18n'
+    import { NftActivity } from '@core/wallet'
+    import { parseLayer2Metadata } from '@core/layer-2'
+
+    export let activity: NftActivity
```
Comment (MarkNerdi) [inline packages/shared/components/molecules/activity-info/SmartContractActivityInformation.svelte:7]: i suppose this should be either `Activity` if all are allowed or `BasicActivity` if only normal transactions are allowed?

**Review by MarkNerdi — DISMISSED** [review]
LGMT, 2 small comments to address

File: packages/shared/components/molecules/activity-info/SmartContractActivityInformation.svelte (hunk 2 above)
Comment (Tuditi) [reply packages/shared/components/molecules/activity-info/SmartContractActivityInformation.svelte:7]: For now only `BasicActivity` is allowed. In the future we will allow more actvities.

File: packages/shared/components/molecules/activity-info/SmartContractActivityInformation.svelte (hunk 1 above)
Comment (Tuditi) [reply packages/shared/components/molecules/activity-info/SmartContractActivityInformation.svelte:22]: I disagree. Ethereum address should already show up in the top-level destination. So there is no need to duplicate this within smart contract details.

Comment (MarkNerdi) [reply packages/shared/components/molecules/activity-info/SmartContractActivityInformation.svelte:22]: thats true, i was just thinking until we're at that state, we can add it here. Currently the contract address is still shown, right?

Comment (Tuditi) [reply packages/shared/components/molecules/activity-info/SmartContractActivityInformation.svelte:22]: It's already the Ethereum address for the send confirmation popup, but it's not shown in the activity information. This has to be addressed in this PR

File: packages/shared/lib/core/layer-2/utils/parseLayer2Metadata.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        }
+        return parsedMetadata
+    } catch (error) {
+        return undefined
+    }
+}
+
+function validate(data: Layer2Metadata): void {
+    if (data.senderContract && typeof data.senderContract !== 'string') {
+        throw 'Invalid senderContract'
```
Comment (Tuditi) [inline packages/shared/lib/core/layer-2/utils/parseLayer2Metadata.ts:25]: Let's follow the official syntax and only throw Errors. 
```suggestion
        throw new Error('Invalid senderContract')
```

File: packages/shared/components/molecules/activity-info/SmartContractActivityInformation.svelte [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    import { KeyValueBox } from 'shared/components'
+    import { localize } from '@core/i18n'
+    import { TransactionActivity } from '@core/wallet'
+    import { Layer2Metadata } from '@core/layer-2'
+
+    export let activity: TransactionActivity
+
+    $: detailsList = createDetailsList(activity.parsedLayer2Metadata)
+
+    function createDetailsList(metadata: Layer2Metadata): { [key: string]: { data: unknown } } {
```
Comment (Tuditi) [inline packages/shared/components/molecules/activity-info/SmartContractActivityInformation.svelte:11]: I think the data can be of type string as well

File: packages/shared/lib/core/wallet/utils/generateActivity/generateTransactionActivity.ts [hunk 5]
```diff
@@ -54,6 +55,12 @@ export function generateTransactionActivity(
     const sendingInfo = getSendingInformation(processedTransaction, output, account)
     const asyncData = getAsyncDataFromOutput(output, outputId, claimingData, account)
 
+    const sendingAddress = sendingInfo.subject.type === 'address' ? sendingInfo.subject.address : undefined
+    const destinationNetwork = getDestinationNetworkFromAddress(sendingAddress)
+
+    const parsedLayer2Metadata =
+        destinationNetwork !== DestinationNetwork.Shimmer ? parseLayer2Metadata(metadata) : undefined
```
Comment (Tuditi) [inline packages/shared/lib/core/wallet/utils/generateActivity/generateTransactionActivity.ts:62]:
```suggestion
        destinationNetwork === DestinationNetwork.Shimmer ? undefined : parseLayer2Metadata(metadata)
```

**Review by Tuditi — CHANGES_REQUESTED** [review]
Looks good, I found one minor issue however:

The gift storage deposit is appearing
on the smart contract tabs when sending a native token to L2.

image

**Review by Tuditi — APPROVED** [review]
LGTM! :fire:

**Pull request merged**